groups/models.py

```diff
@@ -3,15 +3,29 @@
 from django.db import models
 from django.utils import timezone
 
+from .utils import ChainedQuerySetManager
+
 
 class DiscussionManager(models.Manager):
     def for_group_pk(self, pk):
         return self.get_queryset().filter(group_id=pk)
 
 
-class CommentManager(models.Manager):
+class CommentQuerySet(models.query.QuerySet):
     def for_discussion_pk(self, pk):
-        return self.get_queryset().filter(discussion_id=pk)
+        return self.filter(discussion_id=pk)
+
+    def with_user_may_delete(self, user):
+        """Return a list to avoid removing the added variable with further filters."""
```

Comment:
Good start. I'd also add something to explain that the `queryset` can be manipulated before calling `with_user_may_delete`.